test/promise_fail.js

```diff
@@ -0,0 +1,85 @@
+var tap = require('tap');
+var path = require('path');
+var spawn = require('child_process').spawn;
+var concat = require('concat-stream');
+
+var stripFullStack = require('./common').stripFullStack;
+
+tap.test('callback returning rejected promise should cause that test (and only that test) to fail', function (tt) {
+    tt.plan(1);
+
+    var ps = spawn(process.execPath, [path.join(__dirname, 'promises', 'fail.js')]);
+
+    ps.stdout.pipe(concat(function (rows) {
+        var rowsString = rows.toString('utf8');
+
+        // if the output starts with skip, then skip this test
+        if (/^skip/.test(rowsString)) {
```

Comment:
should this be `/^skip$/`?

Reply:
Added a commit that only runs the test if promises are supported.

Before it would still run the test and print the TAP output after 'skip'